spec/functional/message_with_enum_spec.rb

```diff
@@ -0,0 +1,39 @@
+RSpec.describe "enum message encoding" do
+  before :context do
+    require compile("bacon.proto")
+  end
+
+  after :context do
+    Object.send(:remove_const, :Bacon) if defined?(Bacon)
+    Object.send(:remove_const, :Crunchiness) if defined?(Crunchiness)
+  end
+
+  it "encodes a fatty bacon" do
+    bacon = Bacon.new(:id => 123, :crunchiness => :FATTY)
+    expect(bacon.encode).to eq binary_string("\b{\x10\x00")
+  end
+
+  it "encodes a crunchy bacon" do
+    bacon = Bacon.new(:id => 456, :crunchiness => :CRISPY)
+    expect(bacon.encode).to eq binary_string("\b\xC8\x03\x10\x01")
```

Comment:
these binary strings were generated used the `protobuf` gem so that I would have known good values to work against